__tests__/app.test.js

```diff
@@ -267,7 +271,119 @@ describe("NC-Games app", () => {
+          expect(reviews).toBeSortedBy("designer", {
+            //descending: false,
+            compare: function sort(a, b) {
+              //asc: a-b
+              return a - b;
+            },
+          });
+          expect(reviews).toEqual([
+            {
+              review_id: 2,
+              title: "Jenga",
+              designer: "Leslie Scott",
+              owner: "philippaclaire9",
+              review_img_url:
+                "https://www.golenbock.com/wp-content/uploads/2015/01/placeholder-user.png",
+              review_body: "Fiddly fun for all the family",
+              category: "dexterity",
+              created_at: "2021-01-18T10:01:41.251Z",
+              votes: 5,
+              comment_count: "3",
+            },
+          ]);
+        });
+    });
+    test("Responds with status code 200 and an array of objects of a specific category sorted by 'owner' in descending order and the array has the right number of items.", () => {
+      return request(app)
+        .get(`/api/reviews?sort_by=owner&category=social deduction`)
+        .expect(200)
+        .then(({ body: { reviews } }) => {
+          expect(reviews).toBeSortedBy("owner", {
+            //descending: false,
+            compare: function sort(a, b) {
+              //asc: a-b
+              return a - b;
+            },
+          });
+          reviews.forEach((review) => {
+            expect(review).toEqual(
+              expect.objectContaining({ category: "social deduction" })
+            );
+          });
+          expect(reviews).toHaveLength(11);
+        });
+    });
+    test("Responds with status code 404 for a non-existent category, where category is the only query.", () => {
+      return request(app)
+        .get(`/api/reviews?category=gomoku`)
+        .expect(404)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe(`Category 'gomoku' does not exist.`);
+        });
+    });
+    test("Responds with status code 404 for a non-existent category.", () => {
+      return request(app)
+        .get(`/api/reviews?sort_by=owner&category=gomoku`)
+        .expect(404)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe(`Category 'gomoku' does not exist.`);
+        });
+    });
+    test("Responds with status code 400 for an invalid order query.", () => {
+      return request(app)
+        .get(`/api/reviews?sort_by=owner&order=backwards`)
+        .expect(400)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe(`Order 'backwards' is invalid`);
+        });
+    });
+    test("Responds with status code 400 for an invalid sort_by query.", () => {
+      return request(app)
+        .get(`/api/reviews?sort_by=age&order=backwards`)
+        .expect(400)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe(`Sort_by 'age' is invalid`);
+        });
+    });
```

Comment:
Would love to see a test for if the category query is valid but there are no reviews in that category